src/wallet/wallet_tx_builder.cpp

```diff
@@ -0,0 +1,682 @@
+// Copyright (c) 2022 The Zcash developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or https://www.opensource.org/licenses/mit-license.php .
+
+#include "wallet/wallet_tx_builder.h"
+
+using namespace libzcash;
+
+PrepareTransactionResult WalletTxBuilder::PrepareTransaction(
```

Comment:
Perhaps we should just reorder the functions in the header file.

Reply:
I think to reorder in the header, we’d need to add some forward declarations or something – since this is in `WalletTxBuilder`, which depends on `TransactionEffects`. And I imagine the .cpp was written with the flow in mind of calling `WTB::PrepareTransaction` before we call `TE::ApproveAndBuild`.

I would rather not reorder this at all in the WalletTxBuilder series of PRs, as it’ll likely make a mess of the later PRs that are built on this. If we do a reordering PR at the end of the series, perhaps it could have a reference to this PR so anyone digging through blame has a hint?